pallets/did/src/did_details.rs

```diff
@@ -337,9 +346,17 @@ impl<T: Config> DidDetails<T> {
 			DidError::InputError(InputError::MaxVerificationKeysToRemoveLimitExceeded)
 		);
 
-		if let Some(ref endpoint_url) = update_details.new_endpoint_url {
+		if let DidFragmentUpdateAction::Change(ref service_endpoints) = update_details.service_endpoints_update {
+			ensure!(
+				service_endpoints.urls.len() <= T::MaxEndpointUrlsCount::get().saturated_into::<usize>(),
+				DidError::InputError(InputError::MaxUrlsCountExceeded)
+			);
 			ensure!(
-				endpoint_url.len() <= T::MaxUrlLength::get().saturated_into::<usize>(),
+				// Throws InputError::MaxUrlLengthExceeded if any URL is longer than the max allowed size.
+				!service_endpoints
+					.urls
+					.iter()
+					.any(|url| { url.len() > T::MaxUrlLength::get().saturated_into::<usize>() }),
 				DidError::InputError(InputError::MaxUrlLengthExceeded)
 			);
```

Comment:
Since these checks are the same as in `from_creation_details`, I would like them to be moved to an implementation for `ServiceEndpoints` like so:

In **did_details.rs**
```rust
impl ServiceEndpoints {
	pub fn check_url<T: Config>(&self) -> Result<(), InputError> {
		ensure!(
			self.urls.len() <= T::MaxEndpointUrlsCount::get().saturated_into::<usize>(),
			InputError::MaxUrlsCountExceeded,
		);
		// Throws InputError::MaxUrlLengthExceeded if any URL is longer than the max
		// allowed size.
		ensure!(
			!self
				.urls
				.iter()
				.any(|url| { url.len() > T::MaxUrlLength::get().saturated_into::<usize>() }),
			InputError::MaxUrlLengthExceeded
		);
		Ok(())
	}
}
```

In **errors.rs**
```
impl From<InputError> for DidError {
	fn from(e: InputError) -> Self {
		DidError::InputError(e)
	}
}
```

```suggestion
	service_endpoints.check_url::<T>()?;
```

Reply:
https://github.com/KILTprotocol/mashnet-node/pull/235/commits/39e56899781dd40d33487435b33ccb3909c292d4